### ISSUE TITLE
installation issue/dependency error mongoose 8.0.0

### ISSUE THREAD

**Issue by AnimusXCASH** [mongoose-unique-validator/mongoose-unique-validator#150]
Hi i am doing  open university studies on fullstack, and was wondering if you could provide/is there update coming to support mongoose 8.0.0

```npm install mongoose-unique-validator

npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: blog@1.0.0
npm ERR! Found: mongoose@8.0.0
npm ERR! node_modules/mongoose
npm ERR!   mongoose@"^8.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer mongoose@"^7.0.0" from mongoose-unique-validator@4.0.0
npm ERR! node_modules/mongoose-unique-validator
npm ERR!   mongoose-unique-validator@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

**Comment by AlessioNovi** [comment]
Hi,
Fullstack suggests to stick to version 6.12.2 of mongose.
I tried to install the previous version of mongoose-unique-validator and it seems to work.

Hence you can try to remove the current version of mongoose from your package.json file and then run:
npm install mongoose@6.12.2 --save
Then install the previous version of mongoosse unique validator as below:
npm install --save mongoose-unique-validator@3.1.0

This should do the trick

**Comment by monacodelisa** [comment]
I had the same error yesterday, I downgraded mongoose from 8 to 7.

"mongoose": "^7.0.0",
"mongoose-unique-validator": "^4.0.0"

this seems ok, no more errors, but I haven't tested in the app yet

**Comment by lostmart** [comment]
> Hi, Fullstack suggests to stick to version 6.12.2 of mongose. I tried to install the previous version of mongoose-unique-validator and it seems to work.
> 
> Hence you can try to remove the current version of mongoose from your package.json file and then run: npm install mongoose@6.12.2 --save Then install the previous version of mongoosse unique validator as below: npm install --save mongoose-unique-validator@3.1.0
> 
> This should do the trick

seems to work perfectly ! thanks !

"mongoose": "^6.12.2",
"mongoose-unique-validator": "^3.1.0",

**Comment by viveleroi** [comment]
Mongoose 8 is supported by v5 of this plugin

**Comment by PaulDeUlloa** [comment]
If anyone is here in March 2024,  I had trouble installing _mongoose-unique-validator_ so I tried different versions of mongoose to see if it would then allow me to install _mongoose-unique-validator_ .  Earlier versions were not successful for me.

I had to update my mongoose to, "mongoose": "^8.0.0".

I believe the newly published _mongoose-unique-validator_ "version": "5.0.0", only allows a newer version of mongoose. 

This worked for me, maybe it'll help someone else!!
